packages/sfp-cli/src/core/package/deploymentFilters/FlowVersionFilter.ts

```diff
@@ -0,0 +1,76 @@
+import { ComponentSet, registry } from '@salesforce/source-deploy-retrieve';
+import SFPOrg from '../../org/SFPOrg';
+import QueryHelper from '../../queryHelper/QueryHelper';
+import SFPLogger, { Logger, LoggerLevel } from '@flxblio/sfp-logger';
+import { DeploymentFilter } from './DeploymentFilter';
+import { PackageType } from '../SfpPackage';
+const { XMLBuilder } = require('fast-xml-parser');
+import {  deleteFlows, Flow } from '../../utils/FlowUtils';
+const Table = require('cli-table');
+import { ZERO_BORDER_TABLE } from '../../../ui/TableConstants';
+
+
+export default class FlowVersionFilter implements DeploymentFilter {
+
+  public async apply(org: SFPOrg, componentSet: ComponentSet, logger: Logger): Promise<ComponentSet> {
+
+    let sourceComponents = componentSet.getSourceComponents().toArray();
+    let isFlowFound: boolean = false;
+
+    for (const sourceComponent of sourceComponents) {
+      if (sourceComponent.type.name === registry.types.flow.name) {
+        isFlowFound = true;
+          break;
+      }
+    }
+    if (!isFlowFound) return componentSet;
+
+    try {
+      //count flow versions of each flow definition in the org
+      let query = `SELECT MasterLabel, COUNT(id) RecordCount FROM Flow GROUP BY MasterLabel`;
```

Comment:
do we need this query>

Reply:
I used the query as a pre-filter because it is not possible to use the fetchAll method with query tooling. This can mean that not all versions are found by the limit. (limit 2000) On our Org we have over 10000 flow versions. So there are two possibilities. a.) my prefilter via the count to exclude a few flows or a loop on the tooling query via the next url. You have to know that in my solution we have not only deleted the last version, but the wish was to specify the number of flows that should remain via a flag. This flag meant that there were more flows to be deleted. I also see this flag as necessary. So there were more flow versions to choose from. And I didn't want to strain the Org with individual selects.

Reply:
gotcha, thanks